### ISSUE TITLE
NamedTuple functional form doesn't allow literal expressions to be used for field names

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6878]
The [typing spec](https://typing.readthedocs.io/en/latest/spec/qualifiers.html#semantics-and-examples) says:

> Type checkers should treat uses of a final name that was initialized with a literal as if it was replaced by the literal. For example, the following should be allowed:

```python
from typing import NamedTuple, Final

X: Final = "x"
Y: Final = "y"
N = NamedTuple("N", [(X, int), (Y, int)])
```

Currently this doesn't work in pyright.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.345, which I just published. It will also be included in a future release of pylance.
